roles/parse_operator_bundle/tasks/main.yml

```diff
@@ -1,104 +1,137 @@
-  shell: "{{ yq_bin_path }} read --tojson {{ csv_path }} 'spec' | {{ jq_bin_path }} '.installModes[] | select(.type==\"OwnNamespace\").supported'"
-  register: operator_ownnamespace_support_result
-
-- name: "Check if operator install mode supports SingleNamespace"
-  shell: "{{ yq_bin_path }} read --tojson {{ csv_path }} 'spec' | {{ jq_bin_path }} '.installModes[] | select(.type==\"SingleNamespace\").supported'"
-  register: operator_singlenamespace_support_result
-
-- name: "Check if operator install mode supports MultiNamespace"
-  shell: "{{ yq_bin_path }} read --tojson {{ csv_path }} 'spec' | {{ jq_bin_path }} '.installModes[] | select(.type==\"MultiNamespace\").supported'"
-  register: operator_multinamespace_support_result
-
-- set_fact:
-    operator_allnamespaces_support: "{{ operator_allnamespaces_support_result.stdout if operator_allnamespaces_support_result.stdout != '' else false }}"
-    operator_ownnamespace_support: "{{ operator_ownnamespace_support_result.stdout if operator_ownnamespace_support_result.stdout != '' else false }}"
-    operator_singlenamespace_support: "{{ operator_singlenamespace_support_result.stdout if operator_singlenamespace_support_result.stdout != '' else false }}"
-    operator_multinamespace_support: "{{ operator_multinamespace_support_result.stdout if operator_multinamespace_support_result.stdout != '' else false }}"
-
-- name: "Output all collected data to a yaml file in work dir"
-  template:
-    src: "parsed_operator_data.yml.j2"
-    dest: "{{ work_dir }}/parsed_operator_data.yml"
-
-- name: "Sanity check the operator bundle's information"
-  include_tasks: "bundle_sanity_checks.yml"
-  when:
-    bundle_sanity_checks|bool
\ No newline at end of file
+- name: "Parsing the operator bundle data"
+  block:
+  - name: "Inspect the bundle image with skopeo"
+    shell: "skopeo inspect docker://{{ bundle_image }}"
+    register: skopeo_inspect_result
+
+  - name: "Save the skopeo inspect output to a log file"
+    copy:
+      content: "{{ skopeo_inspect_result.stdout }}"
+      dest: "{{ work_dir }}/bundle-skopeo-inspect.json"
+    when:
+      - skopeo_inspect_result is defined
+      - skopeo_inspect_result.stdout != ""
+
+  - set_fact:
+      skopeo_inspect_json: "{{ skopeo_inspect_result.stdout | from_json }}"
+
+  - set_fact:
+      is_bundle_image: "{{ skopeo_inspect_json.Labels['com.redhat.delivery.operator.bundle'] }}"
+      package_name: "{{ skopeo_inspect_json.Labels['operators.operatorframework.io.bundle.package.v1'] }}"
+      default_channel: "{{ skopeo_inspect_json.Labels['operators.operatorframework.io.bundle.channel.default.v1'] }}"
+      channels: "{{ skopeo_inspect_json.Labels['operators.operatorframework.io.bundle.channels.v1'].split(',') }}"
+      ocp_versions: "{{ skopeo_inspect_json.Labels['com.redhat.openshift.versions'].split(',') }}"
+      # if backport label missing, an empty is_backport means not set.
+      is_backport: ""
+
+  - set_fact:
```

Comment:
@dirgim NIT:  Naming every task would be best practice and standard. 
In this case, there might be a chance the task might fail. 
If that channels variable is empty.
It would be great if we could have a name for this task
like, 
```
name: "fetch the first channel from the channels list"
```.

Reply:
Ack, great point - fixed.